src/main/java/algorithms/binarySearch/binarySearchTemplated/README.md

```diff
@@ -0,0 +1,119 @@
+# BinarySearchTemplated
+
+## Background
+
+BinarySearchTemplated is a more generalised algorithm of [BinarySearch](../binarySearch) that removes the condition that
+checks if the current mid-value is equal to the target (which helps to end the search the moment the target is found). 
+The template adds a "condition" method which will be modified based on the requirements of the implementation.
+
+The narrowing of the search space differs from BinarySearch - only the high pointer will be adjusted by one unit.
+
+This template will work for most binary search problems and will require the following changes when used for different 
+problems:
+- Search space (high and low pointers)
+- Condition method
+- Returned value (low or low - 1)
+
+### Search Space (Requires change)
+Initialise the boundary values of the high and low pointers to include all possible elements in the search space.
+
+### Condition (Requires change)
+We assume that when the condition returns true, the current value "passes" and when the condition returns false, the
```

Comment:
maybe emphasize here we're finding the first "P" here.

i think this was also mentioned but in case it isnt:
Sometimes, the problem may require to find the largest value such that it doesnt meet some criteria (ie last "fail"). in this set-up, it is just the index before the first "P"